src/utils/thread_helper.cpp

```diff
@@ -1392,26 +1429,11 @@ otError ThreadHelper::RetrieveTelemetryData(Mdns::Publisher *aPublisher, threadn
 #endif // OTBR_ENABLE_TREL
 
 #if OTBR_ENABLE_BORDER_ROUTING
-        // Begin of InfraLinkInfo section.
-        {
-            auto                           infraLinkInfo = wpanBorderRouter->mutable_infra_link_info();
-            otSysInfraNetIfAddressCounters addressCounters;
-            uint32_t                       ifrFlags = otSysGetInfraNetifFlags();
-
-            otSysCountInfraNetifAddresses(&addressCounters);
-
-            infraLinkInfo->set_name(otSysGetInfraNetifName());
-            infraLinkInfo->set_is_up((ifrFlags & IFF_UP) != 0);
-            infraLinkInfo->set_is_running((ifrFlags & IFF_RUNNING) != 0);
-            infraLinkInfo->set_is_multicast((ifrFlags & IFF_MULTICAST) != 0);
-            infraLinkInfo->set_link_local_address_count(addressCounters.mLinkLocalAddresses);
-            infraLinkInfo->set_unique_local_address_count(addressCounters.mUniqueLocalAddresses);
-            infraLinkInfo->set_global_unicast_address_count(addressCounters.mGlobalUnicastAddresses);
-        }
-        // End of InfraLinkInfo section.
+        // InfraLinkInfo section.
+        RetrieveInfraLinkInfo(*wpanBorderRouter->mutable_infra_link_info());
 
         // ExternalRoutes section
-        RetrieveExternalRouteInfo(wpanBorderRouter->mutable_external_route_info());
+        RetrieveExternalRouteInfo(*wpanBorderRouter->mutable_external_route_info());
```

Comment:
nit: I think the method names are clear enough so maybe you don't have to document them in comments.

Reply:
updated, thanks